File: Javascript_101/Fibonacci.js
var fibonacci = function(max){
    var fib_nums = [];
    var i = 1;
    var j = 2;
    var sum = 0;

    fib_nums.push(i);

    while(!(j>max)){
        fib_nums.push(j);
        var next_fib = i + j;
        i = j;
        j = next_fib;
    }


    for(var count = fib_nums.length; count--;){
        if(fib_nums[count] % 2 === 0){
            sum += fib_nums[count];
        }
    }
    return sum;
}

fibonacci(4000000)

File: Javascript_101/Fibonacci.rb
def evenFibonacciSum(max)
  i = 1
  j = 2
  fibonacci = []
  fibonacci << i
  sum = 0

  while !(j > max)
    fibonacci << j
    next_fib = i + j
    i = j
    j = next_fib
  end


  fibonacci.each do |fib_num|
    if fib_num % 2 == 0
      sum += fib_num
    end
  end

  sum
end

puts evenFibonacciSum(4000000)